cont3xt/vueapp/src/components/integrations/IntegrationTable.vue

```diff
@@ -250,37 +254,88 @@ export default {
 
       if (!newSearchTerm) {
         this.filteredData = this.data;
+        this.highlightData = null;
         this.setTableLen();
         syncWithParent();
         return;
       }
-      this.filteredData = this.data.filter((row) => {
-        let match = false;
-        const query = newSearchTerm.toLowerCase();
 
-        for (const field of this.fields) {
+      /**
+       * @param arr the array to be simultaneously filtered/mapped over
+       * @param filterMapFunc (arrElement) -> [boolean, outputElement]
+       */
+      const filterMap = (arr, filterMapFunc) => {
+        const outputArr = [];
+        for (const elem of arr) {
+          const [passedFilter, outputElement] = filterMapFunc(elem);
+          if (passedFilter) {
+            outputArr.push(outputElement);
+          }
+        }
+        return outputArr;
+      };
+
+      // filters while mapping to avoid needing to re-match values for highlighting data
+      const filterAndHighlightData = filterMap(this.data, (row) => {
+        /* helpers ----------- */
+        const cellHighlightDataFromMatch = (matchObj) => {
+          const matchStart = matchObj.index;
+          // create span information for highlighting
+          return [{ start: matchStart, end: matchStart + matchObj[0].length }];
+        };
+        const matchAndCellHighlightDataFromValue = (fullValue, matchQuery) => {
+          let match = false;
+          let cellHighlightData = null;
+          if (Array.isArray(fullValue)) {
+            cellHighlightData = [];
+            for (const [index, value] of fullValue.entries()) {
+              const matchObj = value.toString().toLowerCase().match(matchQuery);
```

Comment:
if we are truly doing RegExp then you should convert matchQuery to once before calling match, otherwise match will do for every call. If not supporting regexp should this be a indexOf?

Reply:
Good idea! I made the change to create a RegExp beforehand, since regex functionality is desired.